engine/access/rpc/backend/backend.go

```diff
@@ -282,27 +273,32 @@ func executionNodesForBlockID(
 		return flow.IdentityList{}, nil
 	}
 
-	// for now only query Dapperlabs ENs
-	allENs, err := validENs(state)
+	// choose one of the preferred execution nodes
+	subsetENs, err := chooseExecutionNodes(state)
 	if err != nil {
 		return nil, fmt.Errorf("failed to retreive execution IDs for block ID %v: %w", blockID, err)
 	}
 
 	// find the node identities of these execution nodes
-	executionIdentities := allENs.Filter(filter.HasNodeID(executorIDs...))
+	executionIdentities := subsetENs.Filter(filter.HasNodeID(executorIDs...))
 
 	// randomly choose upto maxExecutionNodesCnt identities
 	executionIdentitiesRandom := executionIdentities.Sample(maxExecutionNodesCnt)
 
 	return executionIdentitiesRandom, nil
 }
 
-func validENs(state protocol.State) (flow.IdentityList, error) {
+func chooseExecutionNodes(state protocol.State) (flow.IdentityList, error) {
 	allENs, err := state.Final().Identities(filter.HasRole(flow.RoleExecution))
 	if err != nil {
 		return nil, fmt.Errorf("failed to retreive all execution IDs: %w", err)
 	}
 
+	// if the preferred list of execution ids is not defined, choose from any of the execution node
+	if len(validENMap) == 0 {
```

Comment:
for operators nodes the command line argument will not be defined and they will choose any of the execution nodes